tests/conftest.py

```diff
@@ -27,14 +30,42 @@ def manager():
             os.chdir(oldDirectory)
     return manager
 
-## Command-line option to skip very slow tests
 
 def pytest_addoption(parser):
+    # option to skip very slow tests
     parser.addoption('--fast', action='store_true', help='skip very slow tests')
+    # option to skip parser generation
+    parser.addoption('--skip-pegen', action='store_true', help='generate the parser before running tests')
+
+class ParserGenerationException(BaseException):
+    pass
 
 def pytest_configure(config):
     config.addinivalue_line('markers', 'slow: mark test as very slow')
 
+    if not config.getoption("skip_pegen"):
+        projectRootDir = Path(__file__).parent.parent
+        syntaxDir = projectRootDir / "src" / "scenic" / "syntax"
+        grammar = syntaxDir / "scenic.gram"
+        parser = syntaxDir / "parser.py"
+        result = subprocess.run(
+            [
+                "poetry",
```

Comment:
Is this necessary? If the test suite is being run inside a virtual environment then that version of Python should get used automatically, I think (since the subprocess uses the same `PATH`). This code seems like it could run the subprocess in a different Python than the test suite, if you run the test suite from a virtualenv that isn't managed using Poetry.

Reply:
Fixed faec3f12e3481185a78e9be1eaf7f944ecbbdd31